File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.py
def solution(arr, divisor):
    # arr를 순회하며 divisor로 나누어 떨어지는지를 판단하여 리스트에 넣기
    # result = []
    # for num in arr:
    #     if num % divisor == 0:
    #         result.append(num)
    # 리스트 조건식 이용하기
    result = [x for x in arr if x % divisor == 0]
    # 리스트 오름차순 정렬
    result.sort()
    # 리스트가 비었다면 -1담아 리턴
    if not result:
        result.append(-1)
    return result

File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public static void log(Object o){
        System.out.println(o);
    }
    public int[] solution(int[] arr, int divisor) {
        int[] filteredArr = Arrays.stream(arr)
            .filter(n -> n % divisor == 0)
            .toArray();
        Arrays.sort(filteredArr);
        
        return filteredArr.length == 0 
            ? new int[]{-1} 
            : filteredArr;
    }
}